tests/testImageDecorrelation.py

```diff
@@ -190,65 +182,124 @@ def setUp(self):
+        statObj = afwMath.makeStatistics(diffExp.getMaskedImage(), afwMath.VARIANCECLIP,
+                                         self.statsControl)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        self.assertNotClose(var, expected_var, rtol=0.02)
 
-        diffExp = matched_im1ex.clone()
-        tmpArr = diffExp.getMaskedImage().getImage().getArray()
-        tmpArr -= self.im2ex.getMaskedImage().getImage().getArray()
-        # Uncorrected diffim exposure - variance is wrong (too low)
-        self.assertNotClose(tmpArr[~np.isnan(tmpArr)].var(), expected_var, rtol=0.1)
+        # Uncorrected diffim exposure - variance plane is wrong (too low)
+        statObj = afwMath.makeStatistics(diffExp.getMaskedImage().getVariance(),
+                                         diffExp.getMaskedImage().getMask(), afwMath.MEANCLIP,
+                                         self.statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        self.assertNotClose(mn, expected_var, rtol=0.02)
+        print('UNCORRECTED VARIANCE:', var, mn)
 
         task = DecorrelateALKernelTask()
-        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernel)
+        mKernelPsf = measAlg.KernelPsf(mKernel)
+        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernelPsf)
         corrected_diffExp = decorrResult.correctedExposure
 
         # Corrected diffim - variance should be close to expected.
-        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCE,
+        # We set the tolerance a bit higher here since the simulated images have many bright stars
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCECLIP,
                                          self.statsControl)
-        var = statObj.getValue(afwMath.VARIANCE)
-        print(var, expected_var)
-        self.assertClose(var, expected_var, rtol=0.02)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        self.assertClose(var, expected_var, rtol=0.05)
+
+        # Check statistics of variance plane in corrected diffim
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage().getVariance(),
+                                         corrected_diffExp.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP, self.statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        print('CORRECTED VARIANCE:', var, mn)
+        self.assertClose(mn, expected_var, rtol=0.05)
+        self.assertClose(var, mn, rtol=0.05)
+
+    def testDiffimCorrection1(self):
+        """! Test two images with the same variance.
+
+        Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance). Use
+        """
+        self._setUpImages(var1=0.04, var2=0.04)
+        self._testImages()
+
+    def testDiffimCorrection2(self):
```

Comment:
Name the methods something useful so that the test output tells the user what the test does, e.g. "testImageDecorrelation_same_variance", "testImageDecorrelation_higher_science_variance", "testImageDecorrelation_higher_template_variance"